packages/validation-manager/src/ValidationManagerRIP7560.ts

```diff
@@ -25,6 +28,43 @@ export class ValidationManagerRIP7560 implements IValidationManager {
     // TODO
   }
 
+  async _getStakesInfo (operation: OperationBase): Promise<{ senderInfo: StakeInfo, paymasterInfo: StakeInfo, factoryInfo: StakeInfo }> {
+    const senderTMI = await this.stakeManager.getDepositInfo(operation.sender)
+    const senderInfo = {
+      addr: operation.sender,
```

Comment:
Not a good sign if you need to inject `addr` into returned value. Wouldn't it make sense for `_getStakesInfo` to return `Map<string, StakeInfo>`?
The parser would know the roles anyway, why keep them "named" in the return type (`{ senderInfo: StakeInfo, paymasterInfo: StakeInfo, factoryInfo: StakeInfo }`) ? 

Also, anonymous return types become a problem very quickly IMO.

Reply:
the on-chain function return just the data. the bundler objcet returns also the entity name.
using a mapping makes an implicit assumption that the address identifiy the entity.
that is, not only you need the tx.paymaster value to query the paymaster's stake info, but you also rely on the fact it must differ from the other entities.